java/dagger/lint/DaggerKotlinIssuesDetector.kt

```diff
@@ -0,0 +1,238 @@
+                  context.getLocation(annotation),
+                  ISSUE_FIELD_SITE_TARGET_ON_QUALIFIER_ANNOTATION.getBriefDescription(TextFormat.TEXT),
+                  LintFix.create()
+                      .name("Remove 'field:'")
+                      .replace()
+                      .text("field:")
+                      .with("")
+                      .autoFix()
+                      .build()
+              )
+            }
+          }
+        }
+      }
+
+      override fun visitMethod(node: UMethod) {
+        if (!node.isConstructor &&
+            node.hasAnnotation(PROVIDES_ANNOTATION) &&
+            node.hasAnnotation(JVM_STATIC_ANNOTATION)) {
+          val containingClass = node.containingClass?.toUElementOfType<UClass>() ?: return
+          check(containingClass is KotlinUClass)
+          // Smartcast doesn't seem to work in Bazel IntelliJ projects but the below does work
+          if (containingClass.ktClass is KtObjectDeclaration) {
+            val annotation = node.findAnnotation(JVM_STATIC_ANNOTATION)!!
+            context.report(
+                ISSUE_JVM_STATIC_PROVIDES_IN_OBJECT,
+                context.getLocation(annotation),
+                ISSUE_JVM_STATIC_PROVIDES_IN_OBJECT.getBriefDescription(TextFormat.TEXT),
+                LintFix.create()
+                    .name("Remove @JvmStatic")
+                    .replace()
+                    .text("@JvmStatic")
+                    .with("")
+                    .autoFix()
+                    .build()
+            )
+          }
+        }
+      }
+
+      override fun visitClass(node: UClass) {
+        if (node.hasAnnotation(MODULE_ANNOTATION) && node.isCompanionObject(context.evaluator)) {
+          val parent = node.getUastParentOfType<UClass>(false)!!
+          if (parent.hasAnnotation(MODULE_ANNOTATION)) {
+            context.report(
+                ISSUE_MODULE_COMPANION_OBJECTS,
+                context.getLocation(node as UElement),
+                ISSUE_MODULE_COMPANION_OBJECTS.getBriefDescription(TextFormat.TEXT),
+                LintFix.create()
+                    .name("Remove @Module")
+                    .replace()
+                    .text("@Module")
+                    .with("")
+                    .autoFix()
+                    .build()
+
+            )
+          } else {
+            context.report(
+                ISSUE_MODULE_COMPANION_OBJECTS_NOT_IN_MODULE_PARENT,
+                context.getLocation(node as UElement),
+                ISSUE_MODULE_COMPANION_OBJECTS_NOT_IN_MODULE_PARENT.getBriefDescription(TextFormat.TEXT)
+            )
+          }
+        }
+      }
+    }
+  }
+
+
+
+  /** @return whether or not the [this] is a Kotlin `companion object` type. */
+  private fun UClass.isCompanionObject(@Suppress("UNUSED_PARAMETER") evaluator: JavaEvaluator): Boolean {
+    if (this is KotlinUClass && ktClass is KtObjectDeclaration && name == "Companion") {
```

Comment:
`&& name == "Companion"` wouldn't this miss named companion objects?

Reply:
This w as actually a little stale now that lint tools 26.6 were available, cleaned it up in fc3f808